src/control/server/mgmt_system.go

```diff
@@ -193,6 +194,11 @@ func (svc *mgmtSvc) StopRanks(ctx context.Context, req *mgmtpb.RanksReq) (*mgmtp
 	if err != nil {
 		return nil, err
 	}
+
+	// don't publish rank exit events whilst performing controlled shutdown
+	svc.disableEvents(events.RASRankExit)
+	defer svc.enableEvents(events.RASRankExit)
```

Comment:
Do we need to check if these function pointers are unset before attempting to call them?

Reply:
moved the eventPubSub into the mgmtSvc and enable events function called directly now